stl/inc/deque

```diff
@@ -1292,6 +1292,8 @@ public:
         auto _Count = static_cast<size_type>(_Last - _First);
 #endif // _ITERATOR_DEBUG_LEVEL == 2
 
+        if (_Count == 0)
+            return _First;

```

Comment:
We conventionally use braces for all control flow - this helps prevent mistakes during maintenance. (clang-format won't insert braces, otherwise we'd automate this.)

Additionally, we stylistically prefer to separate "non-chained" if-statements with an empty line, so there should be an empty line after the closing brace. That is:

```cpp
if (is_tiger(animal)) {
    take_photo(animal);
}

if (is_cat(animal)) {
    feed(animal);
} else if (is_dog(animal)) {
    pet(animal);
}
```

This convention prevents:

```cpp
if (is_tiger(animal)) {
    take_photo(animal);
} // no empty line, harder to read
if (is_cat(animal)) {
    feed(animal);
} else if (is_dog(animal)) {
    pet(animal);
}
```

from looking like:

```cpp
if (is_tiger(animal)) {
    take_photo(animal);
} else if (is_cat(animal)) { // very different meaning
    feed(animal);
} else if (is_dog(animal)) {
    pet(animal);
}
```